src/transformers/models/pop2piano/feature_extraction_pop2piano.py

```diff
@@ -0,0 +1,400 @@
+                )
+        if beatstep[0] > 0.01:
+            logger.warning(f"`beatstep[0]` is not 0. All beatstep will be shifted by {beatstep[0]}.")
+            beatstep = beatstep - beatstep[0]
+
+        num_steps = self.num_bars * 4
+        num_target_steps = len(beatstep)
+        extrapolated_beatstep = self.extrapolate_beat_times(beatstep, (self.num_bars + 1) * 4 + 1)
+
+        batch = []
+        for i in range(0, num_target_steps, num_steps):
+            start_idx = i
+            end_idx = min(i + num_steps, num_target_steps)
+
+            start_sample = int(extrapolated_beatstep[start_idx] * self.sampling_rate)
+            end_sample = int(extrapolated_beatstep[end_idx] * self.sampling_rate)
+            feature = audio[start_sample:end_sample]
+            batch.append(feature)
+        batch = pad_sequence(batch, batch_first=True, padding_value=self.padding_value)
+
+        return batch, extrapolated_beatstep
+
+    def pad(self, inputs: BatchFeature):
+        """
+        Pads the inputs to same length and returns attention_mask.
+
+        Args:
+            inputs (`BatchFeature`):
+                processed audio features.
+        Return:
+            `BatchFeature` with attention_mask, attention_mask_beatsteps and attention_mask_extrapolated_beatstep added
+            to it:
+            - **attention_mask** numpy.ndarray of shape (batch_size, max_input_features_seq_length) --
+                Example :
+                    1, 1, 1, 0, 0 (audio 1, also here it is padded to max length of 5 thats why there are 2 zeros at
+                    the end indicating they are padded)
+
+                    0, 0, 0, 0, 0 (zero pad to seperate audio 1 and 2)
+
+                    1, 1, 1, 1, 1 (audio 2)
+
+                    0, 0, 0, 0, 0 (zero pad to seperate audio 2 and 3)
+
+                    1, 1, 1, 1, 1 (audio 3)
+            - **attention_mask_beatsteps** numpy.ndarray of shape (batch_size, max_beatsteps_seq_length)
+            - **attention_mask_extrapolated_beatstep** numpy.ndarray of shape (batch_size,
+              max_extrapolated_beatstep_seq_length)
+        """
+
+        input_features_shapes = [input_feature.shape for input_feature in inputs["input_features"]]
+        beatsteps_shapes = [beatsteps.shape for beatsteps in inputs["beatsteps"]]
+        extrapolated_beatstep_shapes = [
+            extrapolated_beatstep.shape for extrapolated_beatstep in inputs["extrapolated_beatstep"]
+        ]
+
+        attention_masks = []
+        for i, input_feature in enumerate(inputs["input_features"]):
+            features_pad_value = max([*zip(*input_features_shapes)][1]) - input_features_shapes[i][1]
+            padding = ((0, 0), (0, features_pad_value), (0, 0))
+            padding_mask = (padding[0], padding[1])
+            zero_array_len = max([*zip(*input_features_shapes)][1])
+
+            input_feature = np.pad(input_feature, padding, "constant", constant_values=self.padding_value)
```

Comment:
Still also not sure why we can't use `self.pad` here? As far as I can tell we're just hand-writing the functionality that we already have in `self.pad`

Reply:
sorry missed it! I will change `np.pad` to `self.pad` it in the upcoming commit.
But still we must keep the [zero array line](https://github.com/susnato/transformers/blob/b7733c4d3757a71aa091738ef4e0aed95391e59e/src/transformers/models/pop2piano/feature_extraction_pop2piano.py#L232) to separate the examples.

Reply:
Okay good with me! Let's leave this thread here so the next reviewer knows :)

Reply:
A comment should be added by the zero array line specifying this so it's clear to anyone who looks at the code. 

Would it be possible to rework `self.pad` to be more general and work for this (and all other) feature extractor(s)

Reply:
I think that even if we modify the `self.pad` to pad inputs with 3 axes, this feature extractor will still need to separate the examples with a zero array line so we still need to change slightly for this model. So I don't think its worth investing the time to change `self.pad` since any other model could come in the future and have a different extractor design from others.